plugins/reference-assets/core/src/assets/choice/types.ts

```diff
@@ -0,0 +1,49 @@
+import type { Asset, AssetWrapper, Binding, ValidationResponse } from '@player-ui/player';
+import type { BeaconDataType } from '@player-ui/beacon-plugin';
+
+/**
+ * A choice asset represents a single selection choice, often displayed as radio buttons in a web context. 
+ * This will allow users to test out more complex flows than just inputs + buttons.
+ */
+export interface ChoiceAsset<AnyTextAsset extends Asset = Asset>
+  extends Asset<'choice'> {
+  /** Context on what choice the user is making */
+  title: AssetWrapper<AnyTextAsset>;
+
+  /** More detailed information about the selection */
+  note?: AssetWrapper<AnyTextAsset>;
+
+  /** The location in the data-model to store the data */
+  binding: Binding;
+
+  /** An array of choices */
+  choices: Array<ChoicesEntry>
+
+  /** Optional additional data */
+  metaData?: {
+    /** Additional data to beacon when this input changes */
+    beacon?: BeaconDataType;
+  };
+}
+
+type ChoicesEntry<AnyTextAsset extends Asset = Asset> = {
```

Comment:
you will need a transformed version of this as well, to populate a property indicating whether or not it is currently selected